### PR TITLE
dnssvcs module was broken after SDK release.

### PR DESCRIPTION
**Pull request by gahlaut-rahul**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'


...
```
## Root Cause
It was because of change in DS in dns SDK.

### CODE REVIEW

**Review by arpit-srivastava-ibm — APPROVED** [review]
This looks perfectly fine

**Comment by hkantare** [comment]
Can you add test results after this change

**Comment by gahlaut-rahul** [comment]
@hkantare  Nothing is changed w.r.t. functionality, build was breaking and it is fixed and tested locally. 
# Below are the logs

```
rahul.gahlautibm.com@Rahuls-MacBook-Pro terraform-provider-ibm % make testacc TEST=./ibm/service/dnsservices 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/dnsservices -v  -timeout 700m
[WARN] Set the environment variable IBM_APPID_TENANT_ID for testing AppID resources, AppID tests will fail if this is not set


[INFO] Set the environment variable SL_REGION for testing ibm_is_region datasource else it is set to default value 'us-south'
[INFO] Set the environment variable SL_ZONE for testing ibm_is_zone datasource else it is set to default value 'us-south-1'
[INFO] Set the environment variable SL_ZONE_2 for testing ibm_is_zone datasource else it is set to default value 'us-south-2'

[INFO] Set the environment variable IS_WIN_IMAGE for testing ibm_is_instance data source else it is set to default value 'r006-d2e0d0e9-0a4f-4c45-afd7-cab787030776'
'instance-01'
[INFO] Set the environment variable IS_BACKUP_POLICY_JOB_ID for testing ibm_is_backup_policy_job datasource
[INFO] Set the environment variable IS_BACKUP_POLICY_ID for testing ibm_is_backup_policy_jobs datasource

[INFO] Set the environment variable IS_SHARE_PROFILE for testing ibm_is_instance resource else it is set to default value 'tier-3iops'
'r006-1cbe9f0a-7101-4d25-ae72-2a2d725e530e'
...
[WARN] Set the environment variable IBM_CODE_ENGINE_RESOURCE_KEY_ID with the ID of a resource key to access a service instance
=== RUN   TestAccIBMPrivateDNSCustomResolverForwardingRulesDataSource_basic
--- PASS: TestAccIBMPrivateDNSCustomResolverForwardingRulesDataSource_basic (257.42s)
=== RUN   TestAccIBMPrivateDNSCustomResolverDataSource_basic
--- PASS: TestAccIBMPrivateDNSCustomResolverDataSource_basic (175.98s)
=== RUN   TestAccIBMPrivateDNSGlbMonitorsDataSource_basic
^CFAIL	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/dnsservices	445.470s
make: *** [testacc] Error 1

rahul.gahlautibm.com@Rahuls-MacBook-Pro terraform-provider-ibm % date
Tue Jul  4 19:49:10 IST 2023
rahul.gahlautibm.com@Rahuls-MacBook-Pro terraform-provider-ibm % 

```

**Comment by gahlaut-rahul** [comment]
@hkantare  
# Do not merge it, we've explored one more issue which is impacting few UTs.

**Review by hkantare — APPROVED** [review]

File: ibm/service/dnsservices/resource_ibm_private_dns_resource_record.go [hunk 1]
```diff
@@ -348,7 +348,7 @@ func resourceIBMPrivateDNSResourceRecordRead(d *schema.ResourceData, meta interf
 	d.Set(pdnsRecordModifiedOn, response.ModifiedOn)
 
 	if *response.Type == "SRV" {
-		data := response.Rdata.(map[string]interface{})
+		data := response.Rdata
 		d.Set(pdnsSrvPort, data["port"])
```
[the remark this answers is not in the document]
Comment (gahlaut-rahul) [reply ibm/service/dnsservices/resource_ibm_private_dns_resource_record.go:352]: Yes, There is no problem but i am fixing some other issue.

**Pull request merged**